docs/guides/spmd_api.ipynb

```diff
@@ -0,0 +1,1135 @@
+{
```

Comment:
<p>"replaced"-&gt;"replaces"</p><p> </p><p>We should perhaps note that there are more complicated/advanced sharding patterns that require annotating -activation- dimension names differently from -parameter- dimension names.  For really advanced stuff, people may wish to do fine-grained manual mesh assignments using the simpler system above, but the logical-naming helpers are useful for getting started in exploring different sharding layouts.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/2730/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Reply:
Added a new section at the bottom of logic axis introduction. Let me know if there could be a better phrasing of the idea!